src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryPath.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.bigquery.connector;
+
+import javax.annotation.Nullable;
+
+/**
+ * BigQuery Path that parses the path in the request of connection service
+ */
+public class BigQueryPath {
+  private String dataset;
+  private String table;
+  private static final int NAME_MAX_LENGTH = 1024;
+  private static final String VALID_NAME_REGEX = "[\\w]+";
+
+  public BigQueryPath(String path) {
+    parsePath(path);
+  }
+
+  private void parsePath(String path) {
+    if (path == null) {
+      throw new IllegalArgumentException("Path should not be null.");
+    }
+
+    //remove heading "/" if exists
+    if (path.startsWith("/")) {
+      path = path.substring(1);
+    }
+
+    // both "" and "/" are taken as root path
+    if (path.isEmpty()) {
+      return;
+    }
+
+    String[] parts = path.split("/", -1);
+
+    // path should contain at most two part : dataset and table
+    // if path ends with / , it's allowed and the last part will be empty string
+    if (parts.length == 3 && !parts[2].isEmpty() || parts.length > 3) {
```

Comment:
same as previous comment, I want to take "//a/b//" as invalid path

Reply:
You can do `path = path.endsWith("/") ? path.substring(0, path.length() - 1) : path` before you split so you don't this complex condition

Reply:
The major reason I do this here without first removing the ending slash is because I want to parse and validate the dataset in the same way even for some special case. I added more comments to make it clear.

Reply:
for "//", the `path.isEmpty()` at line 53 should already return, so you can still get rid of this complex condition.

Reply:
Good catch. I was confused. done :)